### DISCUSSION TITLE
Default workspace for autolaunched windows?

### DISCUSSION THREAD

**Discussion by zkat** [YaLTeR/niri#190] [Q&A] [answered]
Is there any way to configure what workspace a window will end up with when initially spawned? I'd like to be able to configure a bunch of tools at launch and want to auto-organize them into different workspaces instead of having to move stuff around by hand every time I start niri

**Comment by YaLTeR** [comment]
Not yet

**Comment by YaLTeR** [comment, marked as answer]
I'm also not entirely sure yet how this should work. Since from the niri PoV, windows that run at startup are no different from windows that spawn later (I'm not sure it's possible to fully reliably differentiate these two cases).

Once we have window/client rules, we can have a rule for which monitor a window should appear, but that can get a bit weird since later you may want to launch a window on the current monitor and not on the preset monitor.

It might also be possible to have a setting for which monitor becomes focused at startup. I think udev gives us devices right away so we can differentiate the startup case from the later case there.

What would work for you?

**Reply by zkat** [reply]
I really like how Hyprland does it: https://wiki.hyprland.org/Configuring/Window-Rules/, which, yeah, applies to every time you open an application, but I think that's totally fine. I think opening a window _later_ and having to move it is totally fine tbh.

**Reply by YaLTeR** [reply]
Alright, thanks, that's good then.
